cardinal/ecs/persona_integration_test.go

```diff
@@ -0,0 +1,267 @@
+	// originally got the CreatePersona.
+	tick := world.CurrentTick()
+	_, err = world.GetSignerForPersonaTag(personaTag, tick)
+	assert.ErrorIs(t, err, ecs.ErrCreatePersonaTxsNotProcessed)
+
+	assert.NilError(t, world.Tick(ctx))
+	// The CreatePersona has now been processed
+	addr, err := world.GetSignerForPersonaTag(personaTag, tick)
+	assert.NilError(t, err)
+	assert.Equal(t, addr, signerAddress)
+}
+
+func TestDuplicatePersonaTagsInTickAreOnlyRegisteredOnce(t *testing.T) {
+	world := testutils.NewTestWorld(t).Instance()
+	assert.NilError(t, world.LoadGameState())
+
+	personaTag := "jeff"
+
+	for i := 0; i < 10; i++ {
+		// Attempt to register many different signer addresses with the same persona tag.
+		ecs.CreatePersonaMsg.AddToQueue(
+			world, ecs.CreatePersona{
+				PersonaTag:    personaTag,
+				SignerAddress: fmt.Sprintf("address_%d", i),
+			},
+		)
+	}
+	tick := world.CurrentTick()
+
+	ctx := context.Background()
+	assert.NilError(t, world.Tick(ctx))
+
+	addr, err := world.GetSignerForPersonaTag(personaTag, tick)
+	assert.NilError(t, err)
+	// Only the first address should be associated with the user
+	assert.Equal(t, addr, "address_0")
+
+	// Attempt to register this persona tag again in a different tick. We should still maintain the original
+	// signer address.
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    personaTag,
+			SignerAddress: "some_other_address",
+		},
+	)
+
+	assert.NilError(t, world.Tick(ctx))
+	addr, err = world.GetSignerForPersonaTag(personaTag, tick)
+	assert.NilError(t, err)
+	// The saved address should be unchanged
+	assert.Equal(t, addr, "address_0")
+}
+
+func TestCreatePersonaFailsIfTagIsInvalid(t *testing.T) {
+	// Verify that the CreatePersona is automatically created and registered with a world.
+	world := testutils.NewTestWorld(t).Instance()
+	assert.NilError(t, world.LoadGameState())
+
+	wantTag := "INVALID PERSONA TAG WITH SPACES"
+	wantAddress := "123_456"
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    wantTag,
+			SignerAddress: wantAddress,
+		},
+	)
+	// This CreatePersona has the same persona tag, but it shouldn't be registered because
+	// it comes second.
+	ecs.CreatePersonaMsg.AddToQueue(
+		world, ecs.CreatePersona{
+			PersonaTag:    wantTag,
+			SignerAddress: "some_other_address",
+		},
+	)
+
+	// PersonaTag registration doesn't take place until the relevant system is run during a game tick.
+	assert.NilError(t, world.Tick(context.Background()))
+
+	count := 0
+	wCtx := ecs.NewWorldContext(world)
+	q, err := wCtx.NewSearch(ecs.Exact(ecs.SignerComponent{}))
+	assert.NilError(t, err)
+	err = q.Each(
+		wCtx, func(id entity.ID) bool {
+			count++
+			sc, err := ecs.GetComponent[ecs.SignerComponent](wCtx, id)
+			assert.NilError(t, err)
+			assert.NotEqual(t, sc.PersonaTag, wantTag)
+			assert.NotEqual(t, sc.SignerAddress, wantAddress)
+			return true
```

Comment:
all we're checking is that no signer components exist. so this code should just do the count++ to ensure it didnt increment.